### PR TITLE
FOUR-12992: The permission View Process Catalog, needs to enable per Default for the new users

### PR DESCRIPTION
**Pull request by pmPaulis**
## Issue & Reproduction Steps
The permission View Process Catalog, needs to enable per Default for the new users

## How to Test
Describe how to test that this solution works.

## Related Tickets & Packages
- https://processmaker.atlassian.net/browse/FOUR-12992

## Code Review Checklist
- [ ] I have pulled this code locally and tested it on my instance, along with any associated packages.
- [ ] This code adheres to [ProcessMaker Coding Guidelines](https://github.com/ProcessMaker/processmaker/wiki/Coding-Guidelines).
- [ ] This code includes a unit test or an E2E test that tests its functionality, or is covered by an existing test.
- [ ] This solution fixes the bug reported in the original ticket.
- [ ] This solution does not alter the expected output of a component in a way that would break existing Processes.
- [ ] This solution does not implement any breaking changes that would invalidate documentation or cause existing Processes to fail.
- [ ] This solution has been tested with enterprise packages that rely on its functionality and does not introduce bugs in those packages.
- [ ] This code does not duplicate functionality that already exists in the framework or in ProcessMaker.
- [ ] This ticket conforms to the PRD associated with this part of ProcessMaker.

ci:deploy
ci:next

### CODE REVIEW

**Comment by tokensPM** [comment]
QA server was successfully deployed https://ci-1ace39794f.eng.processmaker.net

**Comment by tokensPM** [comment]
QA server was successfully deployed https://ci-1ace39794f.eng.processmaker.net

**Comment by tokensPM** [comment]
QA server was successfully deployed https://ci-1ace39794f.eng.processmaker.net

File: ProcessMaker/Observers/UserObserver.php [hunk 1]
```diff
@@ -28,4 +29,15 @@ public function deleting(User $user)
             ->where('user_id', $user->id)
             ->delete();
     }
+
+    /**
+     * Handle the user "created" event.
+     */
+    public function created(User $user): void
+    {
+        $permission = Permission::byName('view-process-catalog')->first();
```
Comment (caleeli) [inline ProcessMaker/Observers/UserObserver.php:38]: Consider to add an upgrade to create the new permission `view-process-catalog`

The code seem to fail, when you send a string the function returns the Permission object and you do not need to call the method `first()`:

```suggestion
        $permission = Permission::byName('view-process-catalog')
```

**Review by caleeli — CHANGES_REQUESTED** [review]

**Comment by tokensPM** [comment]
QA server was successfully deployed https://ci-1ace39794f.eng.processmaker.net

**Comment by tokensPM** [comment]
QA server was successfully deployed https://ci-1ace39794f.eng.processmaker.net

**Comment by tokensPM** [comment]
QA server was successfully deployed https://ci-1ace39794f.eng.processmaker.net

**Comment by tokensPM** [comment]
QA server was successfully deployed https://ci-1ace39794f.eng.processmaker.net

**Comment by pmPaulis** [comment]
@caleeli 
I reviewed the functionality, please see the video

https://github.com/ProcessMaker/processmaker/assets/42388723/ac284e87-0746-4650-a1d3-f43e54c46a95

**Comment by tokensPM** [comment]
QA server was successfully deployed https://ci-1ace39794f.eng.processmaker.net

**Comment by tokensPM** [comment]
QA server was successfully deployed https://ci-1ace39794f.eng.processmaker.net

File: upgrades/2023_12_21_142322_create_permission_view_process_catalog.php [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+use ProcessMaker\Upgrades\UpgradeMigration as Upgrade;
+
+class CreatePermissionViewProcessCatalog extends Upgrade
+{
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        if (!Permission::where('name', 'view-process-catalog')->first()) {
```
Comment (caleeli) [inline upgrades/2023_12_21_142322_create_permission_view_process_catalog.php:13]: Consider to use:
```
Permission::updateOrCreate(...)
```

File: upgrades/2023_12_21_142322_create_permission_view_process_catalog.php [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+            ]);
+        }
+    }
+
+    /**
+     * Reverse the migrations.
+     */
+    public function down(): void
+    {
+        if ($permission = Permission::where('name', 'view-process-catalog')->first()) {
```
Comment (caleeli) [inline upgrades/2023_12_21_142322_create_permission_view_process_catalog.php:27]: Consider to use:
```
Permission::where('name', 'view-process-catalog')->delete();
```

**Review by caleeli — APPROVED** [review]

**Comment by tokensPM** [comment]
QA server was successfully deployed https://ci-1ace39794f.eng.processmaker.net

**Comment by tokensPM** [comment]
QA server was successfully deployed https://ci-1ace39794f.eng.processmaker.net

**Comment by tokensPM** [comment]
QA server was successfully deployed https://ci-1ace39794f.eng.processmaker.net

**Comment by tokensPM** [comment]
QA server was successfully deployed https://ci-1ace39794f.eng.processmaker.net

**Pull request merged**